tests/utils/utils.go

```diff
@@ -0,0 +1,333 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	warpBackend "github.com/ava-labs/subnet-evm/warp"
+
+	"github.com/ava-labs/awm-relayer/messages/teleporter"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/interfaces"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/tests/utils"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	predicateutils "github.com/ava-labs/subnet-evm/utils/predicate"
+	"github.com/ava-labs/subnet-evm/x/warp"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	DefaultTeleporterTransactionGas       uint64 = 200_000
+	DefaultTeleporterTransactionGasFeeCap        = big.NewInt(225 * params.GWei)
+	DefaultTeleporterTransactionGasTipCap        = big.NewInt(params.GWei)
+	DefaultTeleporterTransactionValue            = common.Big0
+)
+
+type SendCrossChainMessageEvent struct {
+	DestinationChainID ids.ID
+	MessageID          *big.Int
+	Message            teleporter.TeleporterMessage
+}
+
+type ReceiveCrossChainMessageEvent struct {
+	OriginChainID ids.ID
+	MessageID     *big.Int
+	Message       teleporter.TeleporterMessage
+}
```

Comment:
Note: Once we start using the generated abi, the `.go` file comes with structs for events, as well as methods `Parse<eventName>()`.

Reply:
Nice, those will be great to have. It's a shame there aren't similar generated functions for smart contract method inputs and outputs.